quay_search.py

```diff
@@ -0,0 +1,135 @@
+        writer.commit()
+
+    def search_repository(self, p_search_string, p_non_strict):
+        """This function searches matching repositories of an organization.
+         Results are displayed with all available versions. Docker download command is given too."""
+        # with statement closes searcher after usage.
+        with self._index.searcher() as searcher:
+            search_string = u"*"
+            search_string += p_search_string
+            search_string += "*"
+            query = QueryParser("title", self._index.schema).parse(search_string)
+
+            results = searcher.search(query)
+            
+            if p_non_strict:
+                # look for spelling errors and use suggestions as a search term too
+                corrector = searcher.corrector("title")
+                alternative_strings =  corrector.suggest(p_search_string, limit=2)
+
+                # get all repositories with suggested keywords
+                for i in alternative_strings:
+                    search_string = u"*"
+                    search_string += i
+                    search_string += "*"
+                    query = QueryParser("title", self._index.schema).parse(search_string)
+                    results_tmp = searcher.search(query)
+                    results.extend(results_tmp)
+           
+            # get all versions for the found tools
+            dict_results = {}
+            for i in results:
+                additional_info = self.get_additional_repository_information(i['title'])
+                versions = []
+                for j in additional_info:
+                    versions.append(j)
+                dict_results[i['title']] = versions
+            
+            print "The query ", '\033[1m' + p_search_string + '\033[0m', " resulted in", len(results) ,"result(s).",
+            
+            if p_non_strict:
+                print "The search lists the results for ",
+                for i in xrange(len(alternative_strings)):
+                    print '\033[1m' + alternative_strings[i] + '\033[0m', ", ",
+                print "too."
+            for i in dict_results:
+                print i, "\n\t\t\t", 
+                for j in dict_results[i]:
+                    print j, "\t\t", "docker pull quay.io/"+self._organization+"/"+i+":"+j+"\n\t\t\t",
+                print "\n"
+   
+    def get_additional_repository_information(self, p_repository_string):
+        """Function downloads additional information from quay.io to get the tag-filed which includes the version number."""
+        url = 'https://quay.io/api/v1/repository/'
+        url += self._organization
+        url += '/'
+        url += p_repository_string
+        r = requests.get(url, headers={'Accept-encoding': 'gzip'}, timeout=12)
+
+        json_decoder = json.JSONDecoder()
+        decoded_request = json_decoder.decode(r.text)
+        return decoded_request['tags']
```

Comment:
I'm not sure if this is proper Python style<sup>TM</sup>, but I'd support rewriting these lines to:

```python
return json.JSONDecoder().decode(r.text)['tags']
```
What do you say, @bgruening ?